desc/compute/_fast_ion.py

```diff
@@ -354,3 +374,453 @@ def fun(pitch_inv):
+
+
+@register_compute_fun(
+    name="J_alpha",
+    label=(  # ∂_α J_∥ /∫dl = ∮ dl/|v_∥| (v_d ⋅ ∇s) /∫dl, s=ρ²
+        "\\partial_{\\alpha} \\J_{\\parallel}"
+    ),
+    units="~",
+    units_long="m^{-2}",
+    description="Bounce-averaged radial drift.",
+    coordinates="r",
+    dim=1,
+    profiles=[],
+    transforms={"grid": []},
+    params=[],
+    data=[
+        "min_tz |B|",
+        "max_tz |B|",
+        "cvdrift0",
+        "R0",
+        "<v_dot_grads>",
+    ]
+    + Bounce2D.required_names,
+    resolution_requirement="tz",
+    grid_requirement={"can_fft2": True},
+    **_bounce_doc,
+)
+@partial(
+    jit,
+    static_argnames=[
+        "Y_B",
+        "num_transit",
+        "num_well",
+        "num_quad",
+        "num_pitch",
+        "pitch_batch_size",
+        "surf_batch_size",
+        "spline",
+    ],
+)
+def _dJ_dalpha(params, transforms, profiles, data, **kwargs):
+    """Direct measure of omnigenity.
+
+    Exactly equivalent to the bounce-averaged radial drift.
+    """
+    # noqa: unused dependency
+    theta = kwargs["theta"]
+    Y_B = kwargs.get("Y_B", theta.shape[-1] * 2)
+    alpha = kwargs.get("alpha", jnp.array([0.0]))
+    num_transit = kwargs.get("num_transit", 20)
+    num_pitch = kwargs.get("num_pitch", 64)
+    pitch_batch_size = kwargs.get("pitch_batch_size", None)
+    surf_batch_size = kwargs.get("surf_batch_size", 1)
+    assert (
+        surf_batch_size == 1 or pitch_batch_size is None
+    ), f"Expected pitch_batch_size to be None, got {pitch_batch_size}."
+    spline = kwargs.get("spline", True)
+    fl_quad = (
+        kwargs["fieldline_quad"] if "fieldline_quad" in kwargs else leggauss(Y_B // 2)
+    )
+    quad = (
+        kwargs["quad"]
+        if "quad" in kwargs
+        else get_quadrature(
+            leggauss(kwargs.get("num_quad", 32)),
+            (automorphism_sin, grad_automorphism_sin),
+        )
+    )
+
+    def dJ_dalpha0(data):
+        bounce = Bounce2D(
+            grid,
+            data,
+            data["theta"],
+            Y_B,
+            alpha,
+            num_transit,
+            quad,
+            is_fourier=True,
+            spline=spline,
+        )
+
+        # Find the most "leaky"/"lossy" fieldline and pick that,
+        # then integrate over the pitch angle
+        return jnp.sum(
+            jnp.max(jnp.abs(data["radial_drift"]), axis=1)
+            * data["pitch_inv weight"]
+            / data["pitch_inv"] ** 2,
+            axis=-1,
+        ) / bounce.compute_fieldline_length(fl_quad)
+
+    grid = transforms["grid"]
+
+    fourier_transformed_data = {}
+    data["J_alpha"] = _compute(
+        dJ_dalpha0,
+        fourier_transformed_data,
+        data,
+        theta,
+        grid,
+        num_pitch,
+        surf_batch_size,
+        radial_drift=grid.compress(data["<v_dot_grads>"]),
+    )
+    # )--no-verify / (2 * jnp.pi * num_transit * data["R0"])
+    return data
+
+
+@register_compute_fun(
+    name="J_s",
```

Comment:
`adiabatic J_s`?

Also, is there any strong reason to use `s` instead of `rho`? I guess `s` is more common in the literature, but `rho` would make it more consistent with the rest of the code.